api/envoy/api/v2/route/route.proto

```diff
@@ -325,6 +325,15 @@ message RouteMatch {
   message GrpcRouteMatchOptions {
   }
 
+  message CredentialMatchOptions {
+    // If specified, the route will match against whether or not a certificate is presented.
+    google.protobuf.BoolValue presented = 1;
```

Comment:
Nit: only make this a `BoolValue` if the default is not false, otherwise `bool`. I know the API is inconsistent, we will fix this over the next few major versions.

Reply:
I've left these as BoolValue because the default is that they are not defined, so are not matched against. Is there a better way to to define this 'optional' behaviour?